src/lib/molecules/Charts/MapChartD3/MapChartD3.tsx

```diff
@@ -0,0 +1,697 @@
+        context.beginPath();
+        const generateContext = path.context(context);
+        const hoveredRegion = chartData.find((feature) => {
+            generateContext(feature as GeoPermissibleObjects);
+
+            return context.isPointInPath(
+                (x - transformRef.current.x) / transformRef.current.k,
+                (y - transformRef.current.y) / transformRef.current.k
+            );
+        });
+
+        // END Find hovered region
+
+        if (!hoveredRegion) {
+            if (!hoveredRegionRef.current) {
+                return;
+            }
+            hoveredRegionRef.current.properties.brightness = 0;
+            drawMap();
+            hoveredRegionRef.current = null;
+            handleMouseLeave();
+            return;
+        }
+
+        // START Highlight regions depends on brightness
+
+        if (hoveredRegion.properties.value && hoveredRegion.properties.color !== defaultFeatureColor) {
+            if (hoveredRegionRef.current?.id === hoveredRegion.id) {
+                handleMouseEnter(hoveredRegion, event);
+
+                if (hoveredRegionRef.current.properties.brightness === hoveredRegion.properties.brightness) return;
+
+                hoveredRegion.properties.brightness = 0;
+                drawMap();
+            } else {
+                if (hoveredRegionRef.current) hoveredRegionRef.current.properties.brightness = 0;
+
+                hoveredRegion.properties.brightness = brightness;
+                handleMouseEnter(hoveredRegion, event);
+                drawMap();
+            }
+
+            hoveredRegionRef.current = hoveredRegion as IMapChartFeature;
+            onPointOver && onPointOver(hoveredRegion);
+        } else if (
+            hoveredRegionRef.current &&
+            hoveredRegionRef.current.id !== hoveredRegion.id &&
+            hoveredRegionRef.current.properties.value
+        ) {
+            handleMouseLeave();
+            hoveredRegion.properties.brightness = 0;
+            hoveredRegionRef.current.properties.brightness = 0;
+            drawMap();
+            hoveredRegionRef.current = hoveredRegion;
+        }
+        // END Highlight regions depends on brightness
+    };
+
+    const handleClick = (event: PointerEvent) => {
+        if (!contextRef.current || !pathRef.current || !onPointClick) {
+            return;
+        }
+
+        const context = contextRef.current;
+        const path = pathRef.current;
+
+        const [x, y] = pointer(event);
+
+        const clickedRegion = chartData.find((feature) => {
+            context.beginPath();
+            path.context(context)(feature as GeoPermissibleObjects);
+
+            return context.isPointInPath(x, y);
+        });
+        clickedRegion && onPointClick(clickedRegion);
+    };
+
+    // START Draw map
+
+    const drawMap = () => {
+        if (!contextRef.current || !pathRef.current) {
+            return;
+        }
+
+        const context = contextRef.current;
```

Comment:
You can destructure here and use
```suggestion
        const  {save, clearRect, resetTransform, translate .....  and so on}= contextRef.current;
```

Reply:
We can not because when we try to destructure from the context we find errors.